ta/pkcs11/src/processing_symm.c

```diff
@@ -833,3 +833,71 @@ enum pkcs11_rc derive_key_by_symm_enc(struct pkcs11_session *session,
 
 	return rc;
 }
+
+enum pkcs11_rc wrap_data_by_symm_enc(struct pkcs11_session *session,
+				     void *data, uint32_t data_sz,
+				     void *out_buf, uint32_t *out_sz)
+
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	struct active_processing *proc = session->processing;
+	void *in_buf = NULL;
+	uint32_t align = 0;
+	uint32_t in_sz = data_sz;
+	uint32_t tmp_sz = *out_sz;
+	uint8_t *tmp_buf = out_buf;
+
+	switch (proc->mecha_type) {
+	case PKCS11_CKM_AES_ECB:
+	case PKCS11_CKM_AES_CBC:
+		align = data_sz % TEE_AES_BLOCK_SIZE;
+		if (align)
+			in_sz = data_sz + (TEE_AES_BLOCK_SIZE - align);
+
+		if (*out_sz < in_sz) {
+			*out_sz = in_sz;
+			return PKCS11_CKR_BUFFER_TOO_SMALL;
+		}
+
+		if (align) {
+			if (data_sz > TEE_AES_BLOCK_SIZE) {
+				in_sz = data_sz - align;
+				res = TEE_CipherUpdate(proc->tee_op_handle,
```

Comment:
Missing error checking for this.

Reply:
If we get TEE_ERROR_SHORT_BUFFER , I think we can add and assert(0), as this should not happen as we have taken care of that check in line 857. If there is any other error, we should return. I will add a check.